jenkins-agent.ps1

```diff
@@ -108,11 +108,33 @@ if(![System.String]::IsNullOrWhiteSpace($Cmd)) {
 		$Secret = $env:JENKINS_SECRET
 	}
 	$Secret = $Secret.Trim()
-	if(![System.String]::IsNullOrWhiteSpace($Secret)) {
-		$Secret = " $Secret"
-	}
+
+	$ArgumentList = @(
+		$jnlpProtocolOpts,
+		"-cp",
+		"C:/ProgramData/Jenkins/agent.jar",
+		"hudson.remoting.jnlp.Main",
+		"-headless",
+		$Tunnel,
+		$Url,
+		$WorkDir,
+		$Secret,
+		$Name
+	)
 
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
-	#It is fine it blows up for now since it should lead to an error anyway.
-	Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $("$jnlpProtocolOpts -cp C:/ProgramData/Jenkins/agent.jar hudson.remoting.jnlp.Main -headless$Tunnel$Url$WorkDir$Secret $Name")
+
+	Write-Verbose "Starting to execute Jenkins Agent"
+	$process = Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $ArgumentList
+	$exitCode = $process.ExitCode
+	if ($exitCode -eq 0)
```

Comment:
I don't think the process will ever return.

Reply:
Also, without the -PassThru arguement to `Start-Process`, a Process object is not returned:

See https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/start-process?view=powershell-6#outputs